UpdateCosmosMessageView/handler.ts

```diff
@@ -0,0 +1,217 @@
+  typeof RetrievedMessageStatusWithFiscalCode
+>;
+
+interface IStorableError<T> extends Error {
+  readonly body: T;
+  readonly retriable: boolean;
+}
+
+export const storeError = (queueClient: QueueClient) => (
+  storableError: IStorableError<unknown>
+): TE.TaskEither<Error, QueueSendMessageResponse> =>
+  TE.tryCatch(
+    () => queueClient.sendMessage(JSON.stringify(storableError)),
+    E.toError
+  );
+
+export const toStorableError = <T>(body: T) => (
+  error: CosmosErrors | t.Errors | Error
+): IStorableError<T> => ({
+  ...{
+    body,
+    name: "Storable Error"
+  },
+  ...("kind" in error && error.kind === "COSMOS_DECODING_ERROR"
+    ? {
+        message: readableReport(error.error),
+        retriable: false
+      }
+    : "kind" in error && error.kind === "COSMOS_ERROR_RESPONSE"
+    ? {
+        message: JSON.stringify(error.error),
+        retriable: true
+      }
+    : "kind" in error && error.kind === "COSMOS_EMPTY_RESPONSE"
+    ? {
+        message: "Empty cosmos error message",
+        retriable: true
+      }
+    : Array.isArray(error)
+    ? {
+        message: readableReport(error),
+        retriable: false
+      }
+    : {
+        message: error.message,
+        retriable: true
+      })
+});
+
+export const storeAndLogError = <T>(
+  queueClient: QueueClient,
+  telemetryClient: TelemetryClient
+) => (processingError: IStorableError<T>): TE.TaskEither<void, void> =>
+  pipe(
+    processingError,
+    storeError(queueClient),
+    TE.mapLeft(storingError =>
+      telemetryClient.trackEvent({
+        name:
+          "trigger.messages.cqrs.updatemessageview.failedwithoutstoringerror",
+        properties: {
+          processingError: JSON.stringify(processingError),
+          storingError: storingError.message
+        },
+        tagOverrides: { samplingEnabled: "false" }
+      })
+    ),
+    TE.map(() =>
+      telemetryClient.trackEvent({
+        name: "trigger.messages.cqrs.updatemessageview.failed",
+        properties: {
+          processingError: JSON.stringify(processingError)
+        },
+        tagOverrides: { samplingEnabled: "false" }
+      })
+    )
+  );
+
+export const storeAndLogErrorFirst = <T>(
+  queueClient: QueueClient,
+  telemetryClient: TelemetryClient
+) => (error: IStorableError<T>): TE.TaskEither<IStorableError<T>, void> =>
+  pipe(
+    error,
+    storeAndLogError(queueClient, telemetryClient),
+    TE.mapLeft(() => error)
+  );
+
+export const handleStatusChange = (
+  messageViewModel: MessageViewModel,
+  messageModel: MessageModel,
+  blobService: BlobService
+) => (
+  messageStatus: RetrievedMessageStatusWithFiscalCode
+): TE.TaskEither<Error | t.Errors | CosmosErrors, MessageView> =>
+  pipe(
+    messageViewModel.patch(
+      [messageStatus.messageId, messageStatus.fiscalCode],
+      {
+        status: {
+          archived: messageStatus.isArchived,
+          processing: messageStatus.status,
+          read: messageStatus.isRead
+        }
+      }
```

Comment:
Maybe we have to increment version here :)

Reply:
you are right! fixed, thank you!